nsxt/resource_nsxt_policy_predefined_gateway_policy.go

```diff
@@ -0,0 +1,493 @@
+			}
+
+			log.Printf("[DEBUG] Updating Default Rule with ID %s", *rule.Id)
+			return &rule
+		}
+	}
+
+	// This rule is not present in new config - check if was just deleted
+	// If so, the rule needs to be reverted
+	_, oldRules := d.GetChange("default_rule")
+	for _, oldRule := range oldRules.([]interface{}) {
+		oldRuleMap := oldRule.(map[string]interface{})
+		if oldID, ok := oldRuleMap["nsx_id"]; ok {
+			if (rule.Id != nil) && (*rule.Id == oldID.(string)) {
+				rule := revertGatewayPolicyDefaultRule(rule)
+
+				log.Printf("[DEBUG] Reverting Default Rule with ID %s", *rule.Id)
+				return &rule
+			}
+		}
+	}
+
+	return nil
+}
+
+func setPolicyDefaultRulesInSchema(d *schema.ResourceData, rules []model.Rule) error {
+	var rulesList []map[string]interface{}
+	for _, rule := range rules {
+		elem := make(map[string]interface{})
+		elem["description"] = rule.Description
+		elem["log_label"] = rule.Tag
+		elem["logged"] = rule.Logged
+		elem["action"] = rule.Action
+		elem["revision"] = rule.Revision
+		if len(rule.Scope) > 0 {
+			elem["scope"] = rule.Scope[0]
+		}
+		elem["sequence_number"] = rule.SequenceNumber
+		elem["tag"] = initPolicyTagsSet(rule.Tags)
+		elem["path"] = rule.Path
+		elem["nsx_id"] = rule.Id
+
+		rulesList = append(rulesList, elem)
+	}
+
+	return d.Set("default_rule", rulesList)
+}
+
+func revertPolicyPredefinedGatewayPolicy(predefinedPolicy model.GatewayPolicy, m interface{}) (model.GatewayPolicy, error) {
+
+	// Default values for Name and Description are ID
+	empty := ""
+	predefinedPolicy.Description = &empty
+
+	var childRules []*data.StructValue
+
+	for _, rule := range predefinedPolicy.Rules {
+		if rule.IsDefault != nil && *rule.IsDefault {
+			log.Printf("[DEBUG]: Reverting default rule %s", *rule.Id)
+			revertedRule := revertGatewayPolicyDefaultRule(rule)
+			childRule, err := createPolicyChildRule(*revertedRule.Id, revertedRule, false)
+			if err != nil {
+				return model.GatewayPolicy{}, err
+			}
+			childRules = append(childRules, childRule)
+		} else {
+			// Mark for delete all non-default rules
+			log.Printf("[DEBUG]: Deleting rule %s", *rule.Id)
+			childRule, err := createPolicyChildRule(*rule.Id, rule, true)
+			if err != nil {
+				return model.GatewayPolicy{}, err
+			}
+			childRules = append(childRules, childRule)
+		}
+	}
+
+	predefinedPolicy.Rules = nil
+	if len(childRules) > 0 {
+		predefinedPolicy.Children = childRules
+	}
+
+	if len(predefinedPolicy.Tags) > 0 {
+		tags := make([]model.Tag, 0)
+		predefinedPolicy.Tags = tags
+	}
+
+	return predefinedPolicy, nil
+}
+
+func revertGatewayPolicyDefaultRule(rule model.Rule) model.Rule {
+
+	// NOTE: ability to control default action from gateway config is deprecated and discouraged
+	// Deleting default rule in here will not respect force_whitelisting setting on the gateway
+	if len(rule.Tags) > 0 {
+		tags := make([]model.Tag, 0)
+		rule.Tags = tags
+	}
+
+	empty := ""
+	rule.Description = &empty
+	defaultAction := "DROP"
```

Comment:
Use model.Rule_ACTION_DROP?

Reply:
Good point, I double checked and its actually ALLOW :)